### PR TITLE
getting bugsnag errors in batches

### PR DESCRIPTION
**Pull request by akash**
bugsnag has a very stingy rate limit on its api,
this change should keep us within their limits (with the
unfortunate side effect of less up to date numbers, but
better than the current state that just errors and returns us no
data at all)

### CODE REVIEW

File: projects.rb [hunk 1]
```diff
@@ -1,44 +1,88 @@
 class Project
   extend Forwardable
-  def_delegators :@project, :name, :id, :html_url
+
+  PROJECT_BATCHES =
```
Comment (mwagg) [inline projects.rb:4]: It might be easier to maintain if we use Enumerable#each_slice here

File: projects.rb [hunk 2]
```diff
@@ -1,44 +1,88 @@
...
-               open_errors_count: open_errors_count)
+        projects[project[:id]] = self.new(project: project, open_errors_count: open_errors_count)
 
       rescue Bugsnag::Api::ClientError => e
         $stdout.puts "Can't retrieve errors for project #{project.name}, probably due to API rate limit"
       end
-    end.compact
+    end
+    cache = Dalli::Client.new
+    cache.set('last_batch_run', {index: next_batch_to_run, time: 0.seconds.ago})
```
Comment (ansonK) [inline projects.rb]: Does `0.seconds.ago` work outside rails?
Isn't it `Time.now`?

Comment (akash) [reply projects.rb]: crap, no it doesn't! will fix it

File: projects.rb (hunk 1 above)
Comment (akash) [reply projects.rb:4]: not sure that works with the logic? I'm basically loading everything out of the cache and then just updating the ones that are in the next batch to be updated

**Pull request merged**